### PR TITLE
Implement most of Construction draw track preview

### PR DESCRIPTION
**Pull request by duncanspumpkin**
I need this function implemented before the track paint function can be implemented. It is mostly the same as the OpenRCT2 equivalent. 
WiP as some of the backend hasn't been done yet.

### CODE REVIEW

File: src/OpenLoco/Windows/Construction/ConstructionTab.cpp [hunk 1]
```diff
@@ -2472,26 +2475,124 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
     }
 
     // 0x004A0AE5
-    void drawTrack(uint16_t x, uint16_t y, uint16_t selectedMods, uint16_t di, uint8_t trackType, uint8_t trackPieceId, uint16_t colour, uint8_t bh)
+    void drawTrack(const Map::Pos3& pos, uint16_t selectedMods, uint8_t trackType, uint8_t trackPieceId, uint8_t direction, Gfx::Context& context)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/Construction/ConstructionTab.cpp:2478]: Interestingly colour is not used. Perhaps this was leftover from RCT2 development. Instead the colour of the element is based on the company colour.

File: src/OpenLoco/Windows/Construction/ConstructionTab.cpp [hunk 2]
```diff
@@ -2472,26 +2475,124 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
...
+            }
+        }
+
+        const auto& trackPieces = TrackData::getTrackPiece(trackPieceId);
+        const auto trackDirection = direction & 3;
+
+        Map::TileElement backupTileElements[5] = {};
+        Map::TrackElement previewTrackTileElement = {};
+        // Surface, highTypeFlag, 0xF quads, lastTile, 255 base/clear.
+        Map::TileElement previewSideTrackTileElement = *reinterpret_cast<Map::TileElement*>({ 0x80, 0x8F, 255, 255, 0, 0, 0, 0 });
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/Construction/ConstructionTab.cpp:2515]: We need some way to define a tileelement in memory.

**Comment by duncanspumpkin** [comment]
This now only requires #1495 and it is ready to go.

File: src/OpenLoco/Windows/Construction/ConstructionTab.cpp [hunk 3]
```diff
@@ -2472,26 +2475,113 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
...
-        regs.cx = y;
-        regs.edi = selectedMods << 16 | di;
-        regs.bh = bh;
-        regs.edx = colour << 16 | trackPieceId << 8 | trackType;
+        regs.ax = pos.x;
+        regs.cx = pos.y;
+        regs.edi = selectedMods << 16 | pos.z;
+        regs.bh = direction;
+        regs.edx = trackPieceId << 8 | trackType;
         call(0x004A0AE5, regs);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Windows/Construction/ConstructionTab.cpp:2483]: Is this call left in for debugging purposes?

Comment (duncanspumpkin) [reply src/OpenLoco/Windows/Construction/ConstructionTab.cpp:2483]: oh doh i thought i deleted that.

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
Previews aren't working for roads and trams. Train tracks and narrow gauge tracks are previewing fine.

Screenshot (1)
Screenshot (2)

**Comment by duncanspumpkin** [comment]
@AaronVanGeffen that should be that fixed.

**Comment by AaronVanGeffen** [comment]
Confirmed the preview is working fine now. However, I have noticed another bug related to ghost placement. The arrow logic appears to be inverted while hovering over the viewport.

On this branch:
Screenshot (2)
 
Expected:
Screenshot (1)

**Comment by duncanspumpkin** [comment]
> Confirmed the preview is working fine now. However, I have noticed another bug related to ghost placement. The arrow logic appears to be inverted while hovering over the viewport. Not sure if it's due to a change on this branch or a very recent regression, though.
> 
> On this branch: Screenshot (2)
> 
> Expected: Screenshot (1)

Could you try again. I think i just used the wrong constant but due to a refactor i cant quite tell if that is the issue.

**Comment by AaronVanGeffen** [comment]
> Could you try again. I think i just used the wrong constant but due to a refactor i cant quite tell if that is the issue.

Unfortunately, the logic still appears to be inverted.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**